org.eclipse.winery.lsp/src/main/java/org/eclipse/winery/lsp/Server/ServerCore/Completion/CompletionItemGetter.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+package org.eclipse.winery.lsp.Server.ServerCore.Completion;
+
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.winery.lsp.Server.ServerAPI.API.context.LSContext;
+import org.eclipse.winery.lsp.Server.ServerCore.DataModels.ArtifactType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CompletionItemGetter {
+    public List<CompletionItem> getAvailableArtifactTypes(LSContext lsContext) {
+        List<String> artifactTypes = new ArrayList<>();
+        for (Map.Entry<String, ArtifactType> itr:lsContext.getToscaFile().artifactTypes().get().entrySet()) {
+            artifactTypes.add(" " + itr.getKey());
```

Comment:
Just reading the code, I don't get, why a space is prepended. Is this needed to get the artifact types offered when pressing space?

Reply:
The completion triggered when a colon is typed after "derived_from" keyword for example that is why we need add space between the "derived_from:" and the auto completed artifact type to have a result similar to this "derived_from: Root".